src/main/java/io/fabric8/maven/docker/StartMojo.java

```diff
@@ -142,21 +142,14 @@ public synchronized void executeInternal(final ServiceHub hub) throws DockerAcce
                     // Move from waiting to starting status
                     imagesStarting.add(image);
                     imagesWaitingToStart.remove(image);
+                    
+                    if (!startParallel) {
+                        waitForStartedContainer(hub, containerStartupService, startedContainerAliases, imagesStarting);
+                    }
                 }
 
-                // Wait for the next container to finish startup
-                final Future<StartedContainer> startedContainerFuture = containerStartupService.take();
-                try {
-                    final StartedContainer startedContainer = startedContainerFuture.get();
-                    final ImageConfiguration imageConfig = startedContainer.imageConfig;
-
-                    updateAliasesSet(startedContainerAliases, imageConfig.getAlias());
-                    exposeContainerProps(hub.getQueryService(), startedContainer);
-
-                    // All done with this image
-                    imagesStarting.remove(imageConfig);
-                } catch (ExecutionException e) {
-                    rethrowCause(e);
+                if (startParallel) {
+                    waitForStartedContainer(hub, containerStartupService, startedContainerAliases, imagesStarting);
```

Comment:
looks good, I have a synctactic suggestion though: I would use a Future<StartedContainer> as argument instead of constainterStartupService() (so fetching the fututer before this method) and don't feed in `imageStarting` but also handle it outside. Within the loop we then could avoid adding the images to imagesStarting for the non-parallel case when we wait for it to start anyway.

But well, properly a matter of tast, so perfeclty fine for me if we merge it like this.